netstandard/tools/targets/NETStandard.Library.Trimming.targets

```diff
@@ -0,0 +1,74 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="TrimFiles" AssemblyFile="$(NETStandardToolsTaskDirectory)NETStandard.Tools.dll" />
+
+  <Target Name="_determineTrimPackageInputs">
+    <ItemGroup Condition="'$(RootPackageReference)' != 'true'">
+      <TrimFilesRootPackages Include="@(PackageReference)"/>
+    </ItemGroup>
+
+    <!-- todo: move this stuff to NETCore.App package -->
+    <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+      <!-- Native host / CLR for netcoreapp -->
+      <TrimFilesRootFiles Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <TrimFilesRootFiles Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <TrimFilesRootFiles Include="dotnet.exe;dotnet" />
+      <TrimFilesRootFiles Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <TrimFilesRootFiles Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
+
+      <!-- treat platform packages as trimmable -->
+      <TrimmablePackages Include="$(PackageConflictPreferredPackages)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="TrimFilesOnBuild"
+          DependsOnTargets="_determineTrimPackageInputs"
+          AfterTargets="CoreCompile">
+    <ItemGroup>
+      <_trimOtherRuntimeItems Include="@(_LockFileAssemblies)" Exclude="@(_ConflictPackageFiles)" />
+    </ItemGroup>
+
+    <TrimFiles RootFiles="@(IntermediateAssembly);@(TrimFilesRootFiles)"
```

Comment:
The task is actually not very incremental since it needs to feed items into other tasks that aren't incremental.

We could do more work to improve this but @Petermarcu asked me to just get it in without too much polish since its behind opt-in and we want folks to play with it.  Ideas for further improvements: serialize results and reuse if compile output hasn't changed; back reference reading with a cache (that could be shared cross-project).